google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -135,6 +139,8 @@ static final class ClosedException extends RuntimeException {
     this.dbAdminClient = new DatabaseAdminClientImpl(options.getProjectId(), gapicRpc);
     this.instanceClient =
         new InstanceAdminClientImpl(options.getProjectId(), gapicRpc, dbAdminClient);
+    this.databaseAdminClient = createDatabaseAdminClient();
```

Comment:
But doesn't this still create a single `DatabaseAdminClient` that will be shared between everyone that calls `databaseAdminClient()`? Meaning that if there is one place in the code that calls `DatabaseAdminClient#close()`, then it applies to all instances (because they are the same) that have ever been gotten through `databaseAdminClient()`. Even future calls to `databaseAdminClient()` will return a closed instance.

We can do this, but then we should very clearly document the `databaseAdminClient()` method that it returns a shared instance that they should not close, unless they are going to close their application (and which they now need to do in addition to calling `Spanner#close()`)..

Reply:
Based on offline conversations, revised the PR to now create an instance each time an instance is returned. This mitigated most of the edges cases that arise here.